core/src/main/java/org/apache/iceberg/BaseContentScanTask.java

```diff
@@ -95,7 +95,7 @@ public Expression residual() {
   @Override
   public Iterable<ThisT> split(long targetSplitSize) {
     if (file.format().isSplittable()) {
-      if (file.splitOffsets() != null && OFFSET_ORDERING.isOrdered(file.splitOffsets())) {
+      if (file.splitOffsets() != null && OFFSET_ORDERING.isStrictlyOrdered(file.splitOffsets())) {
```

Comment:
Our `OffsetsAwareSplitScanTaskIterator` constructor below called `isStrictlyOrdered` to ensure there are no duplicates. I am just moving this check here as it is being called for each data file being planned. No need to validate it twice as the iterator is package-private and is only used here.